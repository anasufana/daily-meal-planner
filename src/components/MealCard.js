import React from 'react';
import '../css/components/MealResultsListing.css';

class MealCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {   
    };
  }

  render() {
    return (

        <div className="singleColumn">
          <h2 className="mealTitle">Breakfast</h2>
          <div className="imgContainer">
            <img className="imageRecipe" src="/img/healthy-food-1.jpg" alt="Breakfast-recipe" title="Breakfast-recipe"  />
            <a href="#"><div className="hoverBox">
              <h3 className="hoverTitle">Time to cook!</h3>
              <img className="imgCooking" src="/img/cooking.svg" alt="Time" title="Time" height="60" width="60" />
            </div></a>
          </div>
          <div className="mealDetailsContainer">
            <div className="mealDetails">
              <h3 className="recipeTitle">Pastasciutta</h3>
            </div>
            <div className="mealDetails clockDiv">
              <img className="imgClock" src="/img/clock.svg" alt="Time" title="Time" height="20" width="20" />
              <h3 className="recipeDuration">49'</h3>
            </div>
          </div>
        </div>

    );
  }
}

export default MealCard;