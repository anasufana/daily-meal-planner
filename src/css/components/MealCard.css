.single-column {
  border-radius: 5px;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.1);
  background-color: #fff;
  margin-bottom: 20px; }

.meal-details-container {
  display: grid;
  grid-template-columns: 1fr 80px;
  padding: 20px 10px 20px 5px; }

.clock-div {
  position: relative; }

.img-clock {
  left: 20px;
  top: 9px;
  position: absolute; }

.image-recipe {
  width: 100%;
  height: 16vw;
  transition: 0.5s all;
  object-fit: cover; }

.meal-title {
  padding: 10px;
  text-align: left;
  font-size: 1.7rem; }

.recipe-title {
  padding: 10px;
  text-align: left;
  font-size: 1rem; }

.recipe-duration {
  padding: 10px 50px;
  font-size: 1rem;
  border-left: 1px solid rgba(0, 0, 0, 0.3); }

.img-container {
  position: relative;
  overflow: hidden; }

.hover-box {
  position: absolute;
  top: 0;
  left: -100%;
  width: 100%;
  height: 100%;
  background: rgba(255, 255, 255, 0.8);
  transition: all .5s; }

.hover-title,
.img-cooking {
  position: absolute;
  left: 50%;
  top: 50%;
  margin: 0;
  padding: 0;
  transition: all 0.5s;
  transform: translateX(-50%) translateY(-50%); }

.img-cooking {
  top: 63%;
  left: 41%; }

.img-container:hover .hover-box {
  left: 0%;
  cursor: pointer; }

.img-container:hover img {
  transform: scale(1.2);
  transition: all .5s; }

@media screen and (max-width: 769px) {
  .image-recipe {
    height: 60vw; } }
