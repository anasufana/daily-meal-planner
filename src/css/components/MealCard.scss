* {
  margin: 0;
  padding: 0;
}

a {
  text-decoration: none;
}
a:visited,
a:link {
  color: #555;
}
.singleColumn {
  border-radius: 5px;
  box-shadow: 4px 4px 4px rgba(0,0,0,.1);
  background-color: #fff;
  margin-bottom: 20px;
}

.mealDetailsContainer {
  display: grid;
  grid-template-columns: 1fr 80px;
  padding: 20px 10px 20px 5px;
}
.clockDiv {
  position: relative;
}
.imgClock {
  left: 20px;
  top: 9px;
  position: absolute;
}
.imageRecipe {
  width: 100%;
  height: 100%;
  transition: 0.5s all;
}

.mealTitle {
  padding: 10px;
  text-align: left;
  font-size: 1.7rem;
}

.recipeTitle {
  padding: 10px;
  text-align: left;
  font-size: 1rem;
}

.recipeDuration {
  padding: 10px 50px;
  font-size: 1rem;
  border-left: 1px solid rgba(0,0,0,.3);
}

.imgContainer {
  position: relative;
  overflow: hidden;
}

.hoverBox {
  position: absolute;
  top: 0;
  left: -100%;
  width: 100%;
  height: 100%;
  background: rgba(255, 255, 255, 0.8);
  transition: all .5s;
}

.hoverTitle,
.imgCooking {
  position: absolute;
  left: 50%;
  top: 50%;
  margin: 0;
  padding: 0;
  transition: all 0.5s;
  transform: translateX(-50%) translateY(-50%);

}

.imgCooking {
  top: 63%;
  left: 41%;
}

.imgContainer:hover .hoverBox {
  left: 0%;
  cursor: pointer;
}

.imgContainer:hover img {
  transform: scale(1.2);
  transition: all .5s;
}
