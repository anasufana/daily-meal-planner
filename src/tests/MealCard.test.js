/* global it, expect, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import MealCard from '../components/MealCard';

const details = {
  className: 'slide-in-1',
  handleMealRequest: jest.fn(),
  meal: 'Breakfast',
  details: {
    id: 681594,
    image: 'Huevos-Rancheros-681594.jpg',
    imageUrls: ['Huevos-Rancheros-681594.jpg'],
    readyInMinutes: 20,
    title: 'Huevos Rancheros',
  },
};

it('renders correctly', () => {
  const tree = renderer
    .create(<MealCard
      className={details.className}
      meal={details.meal}
      handleMealRequest={details.handleMealRequest}
      details={details.details}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
