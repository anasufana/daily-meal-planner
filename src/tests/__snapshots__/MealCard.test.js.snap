// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="single-column slide-in-1"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <h2
    className="meal-title"
  >
    Breakfast
  </h2>
  <div
    className="img-container"
  >
    <img
      alt="Breakfast-recipe"
      className="image-recipe"
      src="https://spoonacular.com/recipeImages/Huevos-Rancheros-681594.jpg"
      title="Breakfast-recipe"
    />
    <div
      className="hover-box"
    >
      <h3
        className="hover-title"
      >
        Time to cook!
      </h3>
      <img
        alt="Time"
        className="img-cooking"
        height="60"
        src="/img/cooking.svg"
        title="Time"
        width="60"
      />
    </div>
  </div>
  <div
    className="meal-details-container"
  >
    <div
      className="meal-details"
    >
      <h3
        className="recipe-title"
      >
        Huevos Rancheros
         
      </h3>
    </div>
    <div
      className="meal-details clock-div"
    >
      <img
        alt="Time"
        className="img-clock"
        height="20"
        src="/img/clock.svg"
        title="Time"
        width="20"
      />
      <h3
        className="recipe-duration"
      >
        20
        '
      </h3>
    </div>
  </div>
</div>
`;
